File: Java/Java_Basic/src/BreakContinue.java
public class BreakContinue {
    public static void main (String[] args){
        /*Break untuk menghentikan perulangan*/
        var firstValue = 1;

        while (firstValue <= 10){
            System.out.println("Perulangan ke-" + firstValue);
            firstValue++;

            if(firstValue == 6){
                break;
            }
        }

        /*
        * Continue untuk digunakan men-skip perulangan.
        * Jadi nanti perulangan tersebut akan dilewati
        * */

        /*var secondValue = 0;
        do{
            if(secondValue == 8){
                continue; // bila sesuai maka kode dibawah tidak dieksekusi
            }
            secondValue++;
            System.out.println("Perulangan kedua yang ke-" + secondValue);

        }while(secondValue <= 10)*/;

        // contoh 2
        for(var init = 1; init <= 10; init++){
            if(init % 2 == 0){
                continue;
            }

            System.out.println("Perulangan ganjil yang ke-" +  init);
        }
    }
}

File: study/dasar/casting/break_continue.go
package main

import (
	"fmt"
)

func main() {
	for i := 0; i < 10; i++ {
		if i == 5 {
			break
			// perulanngan saat i == 5 maka akan berhenti
		}

		fmt.Println("Perulangan ke-", i)
	}

	for i := 10; i >= 0; i-- {
		if i%2 == 0 {
			continue
			// kode di bawah akan diskip
		}

		fmt.Println("Perulangan ke-", i) // Print yang ganjil
	}
}
